src/aws-resource-mappings.ts

```diff
@@ -256,78 +268,103 @@ export function mapToAwsResource(
-                        heartbeatTimeout: s.heartbeatTimeout,
-                        lifecycleTransition: s.lifecycleTransition,
-                        notificationMetadata: s.notificationMetadata,
-                        notificationTargetArn: s.notificationTargetArn,
-                        roleArn: s.roleArn,
-                        name: s.lifeCycleHookName,
-                    }),
+                        (t: any) =>
+                            <aws.types.input.autoscaling.GroupLaunchTemplate>{
+                                id: t.launchTemplateId,
+                                name: t.launchTemplateName,
+                                version: t.version,
+                            },
+                    ),
+                    initialLifecycleHooks: props.lifecycleHookSpecificationList?.map(
+                        (s: any) =>
+                            <aws.types.input.autoscaling.GroupInitialLifecycleHook>{
+                                defaultResult: s.defaultReason,
+                                heartbeatTimeout: s.heartbeatTimeout,
+                                lifecycleTransition: s.lifecycleTransition,
+                                notificationMetadata: s.notificationMetadata,
+                                notificationTargetArn: s.notificationTargetArn,
+                                roleArn: s.roleArn,
+                                name: s.lifeCycleHookName,
+                            },
+                    ),
                     loadBalancers: props.loadBalancerNames,
                     maxInstanceLifetime: props.maxInstanceLifetime,
                     // mixedInstancesPolicy: FIXME!
                     protectFromScaleIn: props.newInstancesProtectedFromScaleIn,
                     placementGroup: props.placementGroup,
                     serviceLinkedRoleArn: props.serviceLinkedRoleArn,
-                    tags: props.tags?.map((
-                        (m: { key: any; propagateAtLaunch: any; value: any; }) => <aws.types.input.autoscaling.GroupTag>{
-                            key: m.key,
-                            propagateAtLaunch: m.propagateAtLaunch,
-                            value: m.value
-                        })),
-                    targetGroupArns: props.targetGroupARNs,
+                    tags: props.tags?.map(
+                        (m: { key: any; propagateAtLaunch: any; value: any }) =>
+                            <aws.types.input.autoscaling.GroupTag>{
+                                key: m.key,
+                                propagateAtLaunch: m.propagateAtLaunch,
+                                value: m.value,
+                            },
+                    ),
+                    targetGroupArns: props.targetGroupArns,
                     terminationPolicies: props.terminationPolicies,
-                    vpcZoneIdentifiers: props.vPCZoneIdentifier,
+                    vpcZoneIdentifiers: props.vpcZoneIdentifier,
```

Comment:
This and the `targetGroupArns` are the only real changes in this file, the rest is formatting changes due to actually running prettier.